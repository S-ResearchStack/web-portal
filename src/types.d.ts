declare module '@transcend-io/conflux';
declare module 'native-file-system-adapter';

interface ShowSaveFilePickerOptions {
  suggestedName: string;
}

interface Window {
  showSaveFilePicker: (options?: ShowSaveFilePickerOptions) => Promise<FileSystemFileHandle>;
}

declare module '*.svg' {
  import * as React from 'react';

  export const ReactComponent: React.FunctionComponent<
    React.SVGProps<SVGSVGElement> & { title?: string }
  >;

  export default ReactComponent;
}

interface CSSStyleDeclaration {
  webkitOverflowScrolling: string;
}

declare module '*.svg?url' {
  const src: string;
  export default src;
}

declare namespace NodeJS {
  interface ProcessEnv {
    NODE_ENV: 'production' | 'development' | 'test' | 'none';
    API_URL: string;
    SUPERSET_URL: string;
    // TODO: need to move to aws parameter store
    SUPERSET_ID: string;
    SUPERSET_PASSWORD: string;
    GOOGLE_CLIENT_ID: string;
    AUTHENTICATION_MODE: string;
    GOOGLE_OAUTH_URL: string;
    PUBLIC_PATH: string | undefined;
    MOCK_API: string;
    VERSION: string;
  }
}

declare module 'smoothscroll-polyfill' {
  class Polyfill {
    static polyfill(): void;
  }
  export default Polyfill;
}

declare module 'css-spring';

declare module "*.jpg";
declare module "*.png";
