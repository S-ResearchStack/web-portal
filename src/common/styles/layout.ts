import styled from "styled-components";

export const RowContainer = styled.div<{ flex?: number, width?: string, height?: string }>`
  display: flex;
  flex-direction: row;
  flex: ${p => p.flex ?? 1};
  width: ${p => p.width};
  height: ${p => p.height};
  flex-shrink: 0;
`

export const ColumnContainer = styled.div<{ width?: string, height?: string }>`
  display: flex;
  flex-direction: column;
  width: ${p => p.width};
  height: ${p => p.height};
`

export const PaddedColumnContainer = styled(ColumnContainer)`
  padding-left: 8px;
  padding-right: 8px;
`

export const SizedColumnContainer = styled(PaddedColumnContainer)`
  width: 100%;
`

export const SizedColumnScrollContainer = styled(SizedColumnContainer)`
  width: 100%;
  overflow: hidden;
  overflow-y: auto;
`

export const SizedBox = styled.div<{width?: number, height?: number}>`
  width: ${p => p.width ?? 0};
  height: ${p => p.height ?? 0};
  min-width: ${p => p.width ?? 0};
  min-height: ${p => p.height ?? 0};
  max-width: ${p => p.width ?? 0};
  max-height: ${p => p.height ?? 0};
`
