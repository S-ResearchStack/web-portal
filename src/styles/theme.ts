import { css, DefaultTheme } from 'styled-components';
import { px } from './utils';

const colors = {
  // gray gradations
  black: '#000000',
  black60: 'rgba(0, 0, 0, 0.6)',
  black40: 'rgba(0, 0, 0, 0.4)',
  black08: 'rgba(0, 0, 0, 0.08)',
  black15: 'rgba(0, 0, 0, 0.15)',

  // blue gradations
  blue: "#b0c4f1",
  blue40: 'rgba(176, 196, 241, 0.4)',
  blue15: 'rgba(176, 196, 241, 0.15)',
  blue8: 'rgba(176, 196, 241, 0.08)',

  // light mode
  surface: '#FFFFFF',
  background: '#F7F8FA',
  onPrimary: '#FFFFFF',
  backgroundOnPrimary: '#FFFFFF',
  backgroundSurface: '#FFFFFF',
  backgroundOnSurface: '#2D2D2D',
  primary: '#4475E3',
  primaryShadow: '#4475E361',
  primaryLight: '#ECF1FC',
  primaryHovered: '#2F65E0',
  primary05: '#F6F8FE',
  primary10: '#ECF1FC',
  primary20: '#E3EAFB',
  primary30: '#C7D6F7',
  primaryDisabled: '#B3C6F1',
  primaryLightPressed: '#E3EAFB',
  primaryLightFocused: '#C7D6F7',
  textPrimary: '#474747',
  primaryWhite: '#FFFFFF',
  disabled: '#DBDBDB',
  disabled20: 'rgba(219, 219, 219, 0.2)',
  onDisabled: '#A1A1A1',

  textPrimaryBlue: '#4475E3',
  textPrimaryBlue10: 'rgba(68, 117, 227, 0.2)',
  textPrimaryDark: '#474747',
  textSecondaryGray: '#808080',
  textDisabled: '#838383',
  onSurface: '#2D2D2D',
  secondarySkyblue: '#00B0D7', // replace
  backgoundLight: '#F0F0F0',
  primaryBlueHovered: '#2F65E0',
  primaryBluePressed: '#2057D5',

  statusSuccess: '#2DC008',
  statusSuccess10: '#DCF2EA',
  statusSuccessText: '#00825D',

  statusError: '#FF3333',
  statusError10: '#FFEBEB',
  statusErrorText: '#D14343',

  statusWarning: '#F5BF00',
  statusWarning10: '#FFF7D0',
  statusWarningText: '#A36400',

  secondaryViolet: '#944ED7',
  secondaryVioletHovered: '#6A3799',
  secondaryViolet10: '#F4EDFB',

  secondaryTangerine: '#F39C18',
  secondaryTangerineHovered: '#CC8314',
  secondaryTangerine10: '#FEF5E8',

  secondarySkyBlue: '#00B0D7',
  secondarySkyBlueHovered: '#0092B3',
  secondarySkyBlue10: '#E5F7FB',

  secondaryGreen: '#1AD598',
  secondaryGreenHovered: '#16B37F',
  secondaryGreen10: '#E8FBF5',

  secondaryRed: '#FA5F4F',
  secondaryRedHovered: '#CC4E40',
  secondaryRed10: '#FFEFED',

  transparent: 'rgba(0, 0, 0, 0)'
};

export type SpecColorType = keyof typeof colors;

const weightToCssValue = {
  regular: 400,
  upperregular: 450,
  medium: 500,
  semibold: 600,
  bold: 700,
};

const fontSpec = (
  size: number,
  lineHeight: number | string,
  weight: keyof typeof weightToCssValue,
  fontFamily: string
) => css`
  font-family: ${fontFamily}, sans-serif;
  font-size: ${px(size)};
  line-height: ${typeof lineHeight === 'string' ? lineHeight : px(lineHeight)};
  font-weight: ${weightToCssValue[weight]};
  font-style: normal;
`;

const typography = {
  // Inter typeface
  labelRegular: fontSpec(10, 15, 'regular', 'Inter'),
  labelSemibold: fontSpec(10, 15, 'semibold', 'Inter'),
  smallLabelRegular: fontSpec(8, 10, 'regular', 'Inter'),

  bodyXSmallRegular: fontSpec(12, 18, 'regular', 'Inter'),
  bodyXSmallSemibold: fontSpec(12, 18, 'semibold', 'Inter'),
  bodySmallRegular: fontSpec(14, 21, 'regular', 'Inter'),
  bodySmallSemibold: fontSpec(14, 21, 'semibold', 'Inter'),
  bodyMediumRegular: fontSpec(16, 24, 'regular', 'Inter'),
  bodyMediumSemibold: fontSpec(16, 24, 'semibold', 'Inter'),
  bodyLargeRegular: fontSpec(18, 26, 'regular', 'Inter'),
  bodyLargeSemibold: fontSpec(18, 26, 'semibold', 'Inter'),

  headingXSmall: fontSpec(16, 21, 'semibold', 'Inter'),
  headingSmall: fontSpec(18, 23, 'semibold', 'Inter'),
  headingXMedium: fontSpec(18, 27, 'semibold', 'Inter'),
  headingXMediumRegular: fontSpec(18, 27, 'regular', 'Inter'),
  headingMedium: fontSpec(24, 32.2, 'semibold', 'Inter'),
  headingLargeSemibold: fontSpec(32, 42, 'semibold', 'Inter'),
  heading2LargeSemibold: fontSpec(40, 52, 'semibold', 'Inter'),
  headingXLargeSemibold: fontSpec(48, 58, 'semibold', 'Inter'),
  headingXXLargeSemibold: fontSpec(60, 72, 'semibold', 'Inter'),

  portalDisplayLarge: fontSpec(72, 86, 'semibold', 'Inter'),

  sdkBodySmallRegular: fontSpec(14, '130%', 'regular', 'Inter'),
  sdkBodyLargeRegular: fontSpec(18, 24, 'regular', 'Inter'),
  sdkBodyLargeSemibold: fontSpec(18, 24, 'semibold', 'Inter'),
  sdkHeadingMedium: fontSpec(20, 26, 'semibold', 'Inter'),
  sdkBodyXXLRegular: fontSpec(32, 42, 'regular', 'Inter'),
  sdkBodyXXLSemibold: fontSpec(32, '130%', 'semibold', 'Inter'),
  sdkBodyMediumRegular: fontSpec(16, '130%', 'regular', 'Inter'),
  sdkBodyMediumSemibold: fontSpec(16, '130%', 'semibold', 'Inter'),
  sdkDisplayMedium: fontSpec(20, 26, 'semibold', 'Inter'),
  sdkHeadingLarge: fontSpec(40, '130%', 'semibold', 'Inter'),

  // Fira Code typeface
  query14: fontSpec(14, '130%', 'upperregular', 'Fira Code'),
  query12: fontSpec(12, '150%', 'upperregular', 'Fira Code'),
};

const boxShadow = {
  card: `0 0 ${px(2)} ${px(0)} rgba(0,0,0,0.15)`,
  avatar: `0 ${px(8)} ${px(10)} ${px(2)} 0 rgba(0,0,0,0.12)`,
  studyAvatar: `0 ${px(15)} ${px(20)} rgba(180,180,180,0.3)`,
  previewScreen: `0 ${px(2)} ${px(4)} rgba(71,71,71,0.25)`,
  skipLogicHeader: `box-shadow: ${px(3)} ${px(4)} ${px(15)} rgba(0, 0, 0, 0.05)`,
};

const animation = {
  defaultTiming: 'cubic-bezier(0.45, 0.05, 0, 1)',
  rippleTiming: 'cubic-bezier(0.4, 0, 0.2, 1)',
};

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: typeof colors;
    typography: typeof typography;
    boxShadow: typeof boxShadow;
    animation: typeof animation;
  }
}

const theme: DefaultTheme = {
  colors,
  typography,
  boxShadow,
  animation,
};

export default theme;
