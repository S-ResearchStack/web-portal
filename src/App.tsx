import React from 'react';
import { Provider } from 'react-redux';
import { Route, Switch } from 'react-router-dom';
import { ConnectedRouter } from 'connected-react-router';
import { DndProvider } from 'react-dnd';
import { TouchBackend } from 'react-dnd-touch-backend';

import { ThemeProvider } from 'styled-components';

import { store } from 'src/modules/store/store';
import { history, Path } from 'src/modules/navigation/store';
import UserProtectedRoute from 'src/modules/navigation/private-route/UserProtectedRoute';
import MainLayout from 'src/modules/main-layout/MainLayout';
import SignIn from 'src/modules/auth/signin/SignIn';
import SignUp from 'src/modules/auth/signup/SignUp';
import { TooltipProvider, TooltipsList } from 'src/common/components/Tooltip';
import { theme, GlobalStyles } from 'src/styles';

import 'src/modules/auth/authProvider';
import { ModalProvider } from 'src/common/components/Modal';
import buildProvidersTree from './common/components/ProviderTree/ProviderTree';
import RegistrationScreen from 'src/modules/auth/registration/RegistrationScreen';
import TokenProtectedRoute from 'src/modules/navigation/private-route/TokenProtectedRoute';
import CreateStudyScreen from 'src/modules/studies/CreateStudyScreen';


const ProvidersTree = buildProvidersTree([
  [ThemeProvider, { theme: theme }],
  [TooltipProvider, {}],
  [ModalProvider, {}],
  [DndProvider, { backend: TouchBackend, options: { enableMouseEvents: true } }],
  [Provider, { store }]
]);

const App = () => (
  <ProvidersTree>
    <GlobalStyles />
    <ConnectedRouter history={history}>
      <Switch>
        <Route path={Path.SignIn} component={SignIn} />
        <Route path={Path.AccountCreate} component={SignUp} />
        <TokenProtectedRoute path={Path.Registration}>
          <RegistrationScreen />
        </TokenProtectedRoute>
        <UserProtectedRoute path={Path.CreateStudy}>
            <CreateStudyScreen />
        </UserProtectedRoute>
        <UserProtectedRoute>
          <MainLayout />
        </UserProtectedRoute>
      </Switch>
    </ConnectedRouter>
    <TooltipsList />
  </ProvidersTree>
);

export default App;
