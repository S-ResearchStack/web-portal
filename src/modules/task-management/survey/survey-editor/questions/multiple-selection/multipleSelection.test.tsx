import React from 'react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { act, render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from 'styled-components/';
import { TouchBackend } from 'react-dnd-touch-backend';
import { DndProvider } from 'react-dnd';
import { theme } from 'src/styles';
import handler from './index';

/* eslint-disable @typescript-eslint/no-explicit-any */

describe('multiple-selection', () => {
  const expectId = expect.any(String);

  const emptyQuestion = {
    type: 'multiple',
    id: expectId,
    title: '',
    description: '',
    answers: [
      { id: expectId, value: 'Enter option 1' },
      { id: expectId, value: 'Enter option 2' },
    ],
    options: {
      optional: false,
      includeOther: false,
    },
  };

  const notEmptyQuestion = {
    id: 'id',
    type: 'multiple',
    title: 'Title',
    description: 'Description',
    answers: [
      { id: 'survey_question2', value: 'Option 1' },
      { id: 'survey_question3', value: 'Option 2' },
    ],
    options: {
      optional: false,
      includeOther: false,
    },
  };

  const taskItemFromApi = {
    type: 'CHOICE',
    id: 'id',
    title: 'Question Title',
    explanation: 'Question Description',
    required: true,
    itemProperties: {
      options: [{ value: 'Option 1', label: 'Option 1' }, { value: 'Option 2', label: 'Option 2' }],
    },
  };

  it('type should be multiple', async () => {
    expect(handler.type).toEqual('multiple');
  });

  it('[NEGATIVE] should return empty question', async () => {
    expect(handler.createEmpty()).toEqual(emptyQuestion);
  });

  it('isEmpty should return true', async () => {
    expect(handler.isEmpty(emptyQuestion as any)).toBeTrue();
  });

  it('[NEGATIVE] isEmpty should return true if question is undefined', async () => {
    expect(handler.isEmpty(undefined as any)).toBeTrue();
  });

  it('isEmpty should return false', async () => {
    expect(handler.isEmpty(notEmptyQuestion as any)).toBeFalse();
  });

  it('convertFromOtherType should return converted question', async () => {
    const qi = {
      type: 'single',
      id: 'Question1',
      title: 'Title',
      description: 'Description',
      options: {
        optional: false,
        includeOther: false,
      },
      answers: [],
    };
    expect(handler.convertFromOtherType(qi as any)).toEqual({ ...qi, type: 'multiple' });
  });

  it('[NEGATIVE] convertFromOtherType should return empty question', async () => {
    expect(handler.convertFromOtherType({} as any)).toEqual({
      type: 'multiple',
      id: undefined,
      title: undefined,
      description: undefined,
      answers: [
        { id: expectId, value: 'Enter option 1' },
        { id: expectId, value: 'Enter option 2' },
      ],
      options: {
        optional: false,
        includeOther: false,
      },
    });
  });

  it('fromApi should return converted question', async () => {
    expect(handler.fromApi(taskItemFromApi as any)).toEqual({
      type: 'multiple',
      id: expectId,
      title: 'Question Title',
      description: 'Question Description',
      answers: [
        { id: expectId, value: 'Option 1' },
        { id: expectId, value: 'Option 2' },
      ],
      options: {
        optional: false,
        includeOther: false,
      },
    });
  });

  it('[NEGATIVE] fromApi should return undefined if question type is not a choice', async () => {
    expect(
      handler.fromApi({
        ...taskItemFromApi,
        type: '',
      } as any)
    ).toBeUndefined();
  });

  it('toApi should return converted question', async () => {
    expect(handler.toApi(notEmptyQuestion as any)).toEqual({
      id: expectId,
      type: 'CHOICE',
      tag: 'CHECKBOX',
      title: 'Title',
      explanation: 'Description',
      required: true,
      itemProperties: {
        options: [{ value: 'Option 1', label: 'Option 1' }, { value: 'Option 2', label: 'Option 2' }],
      },
    });
  });

  it('[NEGATIVE] toApi should not crash with empty data', async () => {
    expect(handler.toApi({} as any)).toEqual({
      id: undefined,
      type: 'CHOICE',
      tag: 'CHECKBOX',
      title: undefined,
      explanation: undefined,
      required: false,
      itemProperties: {
        options: [],
      },
    });
  });

  it('renderEditorContent should render editor content', async () => {
    const onChange = jest.fn();
    const confirmOptionRemoval = jest.fn();

    const { baseElement, getByTestId, queryAllByTestId } = render(
      <ThemeProvider theme={theme}>
        <DndProvider backend={TouchBackend} options={{ enableMouseEvents: true }}>
          {handler.renderEditorContent({
            surveyId: '',
            question: notEmptyQuestion as any,
            onChange,
            confirmOptionRemoval,
          })}
        </DndProvider>
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();

    const addOptionButton = getByTestId('add-option-button');
    await act(() => userEvent.click(addOptionButton));

    await waitFor(
      () => {
        expect(onChange).toHaveBeenCalled();
      },
      // required because survey input has a debounce for invoking onChange callback
      { timeout: 2500 }
    );

    const firstOptionDeleteButton = queryAllByTestId('delete-option-button')[1];
    await act(async () => userEvent.click(firstOptionDeleteButton));
    await waitFor(() => {
      expect(onChange).toHaveBeenCalled();
    });
  });

  it('[NEGATIVE] renderEditorContent should return null if question is undefined', async () => {
    const onChange = jest.fn();
    const confirmOptionRemoval = jest.fn();

    expect(
      handler.renderEditorContent({
        surveyId: '',
        question: undefined as any,
        onChange,
        confirmOptionRemoval,
      })
    ).toBeNull();
  });

  it('[NEGATIVE] renderEditorContent should return null if question answers is empty', async () => {
    const onChange = jest.fn();
    const confirmOptionRemoval = jest.fn();

    expect(
      handler.renderEditorContent({
        surveyId: '',
        question: { ...notEmptyQuestion, answers: undefined } as any,
        onChange,
        confirmOptionRemoval,
      })
    ).toBeNull();
  });

  it('[NEGATIVE] renderEditorContent should return null if question options is empty', async () => {
    const onChange = jest.fn();
    const confirmOptionRemoval = jest.fn();

    expect(
      handler.renderEditorContent({
        surveyId: '',
        question: { ...notEmptyQuestion, options: undefined } as any,
        onChange,
        confirmOptionRemoval,
      })
    ).toBeNull();
  });
});
