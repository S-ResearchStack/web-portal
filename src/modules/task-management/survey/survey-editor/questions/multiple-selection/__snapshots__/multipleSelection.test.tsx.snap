// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multiple-selection renderEditorContent should render editor content 1`] = `
.c12 {
  position: absolute;
  will-change: opacity,transform;
  border-radius: 50%;
  -webkit-animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  z-index: 0;
  background-color: #E3EAFB;
  pointer-events: none;
}

.c9 {
  position: relative;
  z-index: 0;
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.25rem;
  width: 100%;
  white-space: nowrap;
  padding: 0;
  -webkit-transition-duration: 0;
  transition-duration: 0;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  height: 1.5rem;
  width: 1.5rem;
  border: none;
  background-color: transparent;
}

.c9:hover {
  cursor: pointer;
}

.c9 > div {
  color: #4475E3;
}

.c9 > div svg {
  fill: #4475E3;
}

.c9:hover:enabled > div {
  color: #2F65E0;
}

.c9:hover:enabled > div svg {
  fill: #2F65E0;
}

.c9:hover:enabled:active > div {
  color: #2057D5;
}

.c9:hover:enabled:active > div svg {
  fill: #2057D5;
}

.c9:disabled > div {
  color: #B3C6F1;
}

.c9:disabled > div svg {
  fill: #B3C6F1;
}

.c9:focus-visible {
  outline: 0.125rem solid #C7D6F7;
}

.c9:focus-visible > div {
  color: #4475E3;
}

.c9:focus-visible > div svg {
  fill: #4475E3;
}

.c9:hover:enabled {
  background-color: #ECF1FC;
}

.c9:hover:enabled:active > div {
  color: #2057D5;
}

.c9:hover:enabled:active > div svg {
  fill: #2057D5;
}

.c9:focus-visible:enabled {
  background-color: #C7D6F7;
  outline: none;
}

.c14 {
  position: relative;
  z-index: 0;
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.25rem;
  width: 100%;
  white-space: nowrap;
  padding: 0;
  -webkit-transition-duration: 0;
  transition-duration: 0;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  height: 3rem;
  border: none;
  background-color: transparent;
}

.c14:hover {
  cursor: pointer;
}

.c14 > div {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 600;
  font-style: normal;
}

.c14 > div svg {
  margin-right: 0.25rem;
}

.c14 > div {
  color: #4475E3;
}

.c14 > div svg {
  fill: #4475E3;
}

.c14:hover:enabled > div {
  color: #2F65E0;
}

.c14:hover:enabled > div svg {
  fill: #2F65E0;
}

.c14:hover:enabled:active > div {
  color: #2057D5;
}

.c14:hover:enabled:active > div svg {
  fill: #2057D5;
}

.c14:disabled > div {
  color: #B3C6F1;
}

.c14:disabled > div svg {
  fill: #B3C6F1;
}

.c14:focus-visible {
  outline: 0.125rem solid #C7D6F7;
}

.c14:focus-visible > div {
  color: #4475E3;
}

.c14:focus-visible > div svg {
  fill: #4475E3;
}

.c11 {
  position: relative;
  z-index: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transform: translate3d(0,0,0.0625rem);
  -ms-transform: translate3d(0,0,0.0625rem);
  transform: translate3d(0,0,0.0625rem);
}

.c4 {
  -webkit-scrollbar-width: thin;
  -moz-scrollbar-width: thin;
  -ms-scrollbar-width: thin;
  scrollbar-width: thin;
  -webkit-scrollbar-color: #C7D6F7 #FFFFFF;
  -moz-scrollbar-color: #C7D6F7 #FFFFFF;
  -ms-scrollbar-color: #C7D6F7 #FFFFFF;
  scrollbar-color: #C7D6F7 #FFFFFF;
}

.c4::-webkit-scrollbar {
  width: 1.25rem;
  height: 1.25rem;
}

.c4::-webkit-scrollbar-track {
  background: #FFFFFF;
  border-radius: 0.25rem;
  margin: 0.0625rem 0;
}

.c4::-webkit-scrollbar-thumb {
  background-color: #C7D6F7;
  border-radius: 0.625rem;
  border: 0.5rem solid #FFFFFF;
  width: 0.25rem;
  min-height: 1.25rem;
}

.c5 {
  resize: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  box-sizing: border-box;
  outline: none;
  width: 100%;
  padding: 0;
  margin: 0;
  border: none;
  background: #FFFFFF;
  overflow-y: hidden;
  caret-color: #4475E3;
  overflow-x: hidden;
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #808080;
  min-height: 1.5rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  position: relative;
}

.c3 {
  width: 2.5rem;
  height: 2.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  margin: 0.5625rem 0 0.4375rem !important;
  color: #474747;
}

.c6::-webkit-input-placeholder {
  color: #A1A1A1;
}

.c6::-moz-placeholder {
  color: #A1A1A1;
}

.c6:-ms-input-placeholder {
  color: #A1A1A1;
}

.c6::placeholder {
  color: #A1A1A1;
}

.c8 {
  height: 2.5rem;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
  grid-template-columns: repeat(2,1.5rem);
}

.c10 {
  visibility: hidden;
}

.c13 {
  cursor: move !important;
}

.c1 {
  min-height: 2.5rem;
  height: 2.5rem;
  border-radius: 0.25rem;
  display: grid;
  margin: 0.25rem 0;
  gap: 0.25rem;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  background-color: #FFFFFF;
}

.c2 {
  grid-template-columns: 2.5rem 1fr 4.5rem;
  opacity: 1;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
}

.c2 .c7 {
  opacity: 0;
  -webkit-transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
  transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
}

.c2:hover {
  background-color: #F7F8FA;
}

.c2:hover .c7 {
  opacity: 1;
}

.c15 {
  width: 100%;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  padding-left: 0.5rem;
}

<body>
  <div>
    <div
      class="c0"
    >
      <div
        class="c1 c2"
      >
        <div
          class="c3"
        >
          <svg />
        </div>
        <textarea
          class=" c4 c5 c6"
          maxlength="256"
          placeholder="Enter option"
          rows="1"
          style="overflow-y: hidden; height: 0rem;"
        >
          Option 1
        </textarea>
        <div
          class="c7 c8"
        >
          <button
            aria-label="Delete Option"
            class="c9 c10"
            data-testid="delete-option-button"
            fill="text"
            style="overflow: hidden; position: relative;"
          >
            <div
              class="c11"
              data-testid="content"
            >
              <svg />
            </div>
            <div
              class="c12"
              color="primaryLightPressed"
              opacity="1,1"
              scale="0"
              style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
            />
          </button>
          <button
            aria-label="Drag and Drop Option"
            class="c9 c13"
            fill="text"
            style="overflow: hidden; position: relative;"
          >
            <div
              class="c11"
              data-testid="content"
            >
              <svg />
            </div>
            <div
              class="c12"
              color="primaryLightPressed"
              opacity="1,1"
              scale="0"
              style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
            />
          </button>
        </div>
      </div>
      <div
        class="c1 c2"
      >
        <div
          class="c3"
        >
          <svg />
        </div>
        <textarea
          class=" c4 c5 c6"
          maxlength="256"
          placeholder="Enter option"
          rows="1"
          style="overflow-y: hidden; height: 0rem;"
        >
          Option 2
        </textarea>
        <div
          class="c7 c8"
        >
          <button
            aria-label="Delete Option"
            class="c9 c10"
            data-testid="delete-option-button"
            fill="text"
            style="overflow: hidden; position: relative;"
          >
            <div
              class="c11"
              data-testid="content"
            >
              <svg />
            </div>
            <div
              class="c12"
              color="primaryLightPressed"
              opacity="1,1"
              scale="0"
              style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
            />
          </button>
          <button
            aria-label="Drag and Drop Option"
            class="c9 c13"
            fill="text"
            style="overflow: hidden; position: relative;"
          >
            <div
              class="c11"
              data-testid="content"
            >
              <svg />
            </div>
            <div
              class="c12"
              color="primaryLightPressed"
              opacity="1,1"
              scale="0"
              style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
            />
          </button>
        </div>
      </div>
    </div>
    <button
      class="c14 c15"
      data-testid="add-option-button"
      fill="text"
      style="overflow: hidden; position: relative;"
    >
      <div
        class="c11"
        data-testid="content"
      >
        <svg />
        Add option
      </div>
      <div
        class="c12"
        color="primaryLightPressed"
        opacity="1,1"
        scale="0"
        style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
      />
    </button>
  </div>
</body>
`;
